src/compas/geometry/surfaces/surface.py

```diff
@@ -43,15 +41,41 @@ class Surface(Geometry):
         The parameter domain of the surface in the U direction.
     domain_v : tuple[float, float], read-only
         The parameter domain of the surface in the V direction.
+    is_closed_u : bool, read-only
+        Flag indicating if the surface is closed in the U direction.
+    is_closed_v : bool, read-only
+        Flag indicating if the surface is closed in the V direction.
     is_periodic_u : bool, read-only
         Flag indicating if the surface is periodic in the U direction.
     is_periodic_v : bool, read-only
         Flag indicating if the surface is periodic in the V direction.
+    native_surface : Any, read-only
+        The CAD native surface object.
 
     """
 
+    @property
+    def __data__(self):
+        return {
+            "frame": self._frame.__data__,
+            "domain_u": list(self.domain_u),
+            "domain_v": list(self.domain_v),
+        }
+
+    @classmethod
+    def __from_data__(cls, data):
+        frame = Frame.__from_data__(data["frame"])
+        domain_u = tuple(data["domain_u"])
+        domain_v = tuple(data["domain_v"])
+        return cls.from_plane(frame, domain_u, domain_v)
```

Comment:
since the base surface is abstract and since it should not be possible to create instances of it directly, i don't think it should provide the possibility to create one from data...

Reply:
`from_plane` will invoke the plugin, if available. it just seems to be the canonical surface representation (maybe with exception of `frame` which it's seems I didn't quite get)

Reply:
but why should we provide functionality for making a "canonical" surface from data, when the surface can't actually by instantiated in the normal way. i though we agreed that only the spacial constructors can be used to create an actual surface and that the base class only provides the shared interface...

Reply:
for CAD agnostic (de)serialization? maybe what's confusing me is that lack of other kinds of surface types
maybe this should be in a `PlanarSurface` like in Rhino..

Reply:
there are plenty of other surface types (including planar surface). they just don't require a Rhino or OCC plugin...

Reply:
`compas.geometry.surfaces` has `CylindricalSurface`, `ConicalSurface`, `PlanarSurface`, `SphericalSurface`, `ToroidalSurface`